File: VS2019-C++基础/运算符重载/Person.cpp
#include "Person.h"
#include <iostream>
using namespace std;
Person::Person(const Person& p)
{
	std::cout << "拷贝构造函数" << std::endl;
	this->m_a = p.m_a;
	this->m_b = p.m_b;
}

Person::~Person()
{
	std::cout << "析构函数" << std::endl;
}

void Person::show() const
{
	std::cout << "m_a: " << m_a 
		<< " m_b: " << m_b 
		<< std::endl;
}

const Person operator+(const Person& p1, const Person& p2)
{
	Person temp;
	temp.m_a = p1.m_a + p2.m_a;
	temp.m_b = p1.m_b + p2.m_b;
	return temp;
}

const Person operator-(const Person& p1, const Person& p2)
{
	Person temp;
	temp.m_a = p1.m_a - p2.m_a;
	temp.m_b = p1.m_b - p2.m_b;
	return temp;
}

const Person operator++(const Person& p)
{
	Person temp;
	temp.m_a = p.m_a + 1;
	temp.m_b = p.m_b + 1;
	return temp;
}

File: BookCodeTest/BookCode13/app/src/main/java/com/hqf/bookcode_13/Person.java
package com.hqf.bookcode_13;

import android.widget.Toast;

import java.io.Serializable;

public class Person implements Serializable {//序列话对象，实现简单，效率没有 Parcelable 高
    private long id;
    private String name;
    private String address;

    public Person(long id, String name, String address) {
        this.id = id;
        this.name = name;
        this.address = address;
        Toast.makeText(MyApplication.getContext(), "Hello", Toast.LENGTH_SHORT).show();
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getAddress() {
        return address;
    }
}
